src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java

```diff
@@ -44,137 +29,46 @@
-                    //      On the other hand, a system can react to the OnFullyHealedEvent and unregister the regeneration id if desired.
-                    removeCompleted(currentWorldTime, regen);
-                    if (regen.regenValue.isEmpty()) {
-                        regenEntity.removeComponent(RegenComponent.class);
-                    } else {
-                        regenEntity.saveComponent(regen);
-                        regenSortedByTime.put(findSoonestEndTime(regen), regenEntity);
-                    }
-                });
-
-        // Regenerate the entities with EndTime greater than Current time
-        regenerate(currentWorldTime);
-    }
-
-    private List<EntityRef> getEntitiesWithExpiringRegenIds(long currentWorldTime) {
-        List<EntityRef> entitiesWithExpiringRegenActions = new LinkedList<>();
-        Iterator<Long> regenTimeIterator = regenSortedByTime.keySet().iterator();
-        long endTime;
-        while (regenTimeIterator.hasNext()) {
-            endTime = regenTimeIterator.next();
-            if (endTime == -1) {
-                continue;
-            }
-            if (endTime > currentWorldTime) {
-                break;
-            }
-            entitiesWithExpiringRegenActions.addAll(regenSortedByTime.get(endTime));
-            regenTimeIterator.remove();
-        }
-        return entitiesWithExpiringRegenActions;
-    }
-
-    private void regenerate(long currentTime) {
-        Map<EntityRef, Long> regenToBeRemoved = new HashMap<>();
-        for (EntityRef entity : regenSortedByTime.values()) {
-            RegenComponent regen = entity.getComponent(RegenComponent.class);
-            HealthComponent health = entity.getComponent(HealthComponent.class);
-            if (regen == null || health == null) {
-                logger.debug("Entity '{}' without RegenComponent or HealthComponent scheduled for regeneration - skipping", entity);
-                continue;
-            }
-            if (health.nextRegenTick < currentTime) {
-                health.currentHealth += getRegenValue(regen);
-                health.nextRegenTick = currentTime + 1000;
-                if (health.currentHealth >= health.maxHealth) {
-                    regenToBeRemoved.put(entity, regen.soonestEndTime);
-                    if (hasBaseRegenOnly(regen) || regen.regenValue.isEmpty()) {
-                        entity.removeComponent(RegenComponent.class);
-                    }
-                    entity.send(new OnFullyHealedEvent(entity));
-                }
-                entity.saveComponent(health);
-            }
+        //TODO: bring back to optimization to update each entity only once per second, but scan for
+        //      entities to regenerate every 200ms
+        regenTick += delta;
+        Instant currentTime = Instant.fromMillis(time.getGameTimeInMs());
+
+        if (regenTick > 0.2f) {
```

Comment:
magic number

Reply:
Yep, I'm not yet :100:% sure how to handle the "tick rate" here... But can definitely use a name constant for this :+1: